### PR TITLE
feat(user): add support to delete user

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Add add new endpoint `/user/user/delete` that deletes the user present in the merchant account

Here we have two scenarios
- When user is related to more than one merchant accounts: Delete the user role for that particular merchant account. User will be present in other accounts.
- When user is part of one merchant account: Delete user role and user completely. 

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Currently there is no endpoint to remove the invited users

## How did you test it?
- Singup/Singin
Use the following curl to invite users:
```
curl --location 'http://localhost:8080/user/user/invite' \
--header 'Authorization: Bearer JWT' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "user2@juspay.in",
    "name": "user2",
    "role_id": "merchant_admin"
}

```

Use the following curl to delete invited user:
```
curl --location 'http://localhost:8080/user/user/delete' \
--header 'Authorization: Bearer JWT' \
--header 'Content-Type: application/json' \
--data-raw '{
  "email": "invite_from_test21521@juspay.in"
}'
```

Cases to handle:
- only users with permission UsersWrite can delete
- org_admin cannot be deleted
- internal users cannot be deleted
- user cannot delete himself

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/utils/user.rs [hunk 1]
```diff
@@ -111,3 +112,15 @@ pub fn get_dashboard_entry_response(
         user_role: user_role.role_id,
     })
 }
+
+pub fn can_delete_user_role(role_id: &str) -> UserResult<()> {
+    match role_id {
+        consts::user_role::ROLE_ID_ORGANIZATION_ADMIN
+        | consts::user_role::ROLE_ID_INTERNAL_ADMIN
+        | consts::user_role::ROLE_ID_INTERNAL_VIEW_ONLY_USER
+        | consts::user_role::INTERNAL_USER_MERCHANT_ID => {
```
Comment (racnan) [inline crates/router/src/utils/user.rs:121]: ??

**Review by racnan — CHANGES_REQUESTED** [review]

**Comment by racnan** [comment]
Description indicates you're deleting user from org but looks like you're deleting user from merchant account.

File: crates/router/src/core/user_role.rs [hunk 2]
```diff
@@ -115,3 +117,85 @@ pub async fn update_user_role(
...
+        .list_user_roles_by_user_id(user_from_db.get_user_id())
+        .await
+        .change_context(UserErrors::InternalServerError)?;
+
+    match user_roles
+        .iter()
+        .find(|&role| role.merchant_id == user_from_token.merchant_id.as_str())
+    {
+        Some(user_role) => {
+            utils::user::validate_deletion_permission_for_role_id(&user_role.role_id)?;
```
Comment (racnan) [inline crates/router/src/core/user_role.rs:161]: `user_role` util instead of `user` util

File: crates/router/src/utils/user.rs [hunk 3]
```diff
@@ -111,3 +112,15 @@ pub fn get_dashboard_entry_response(
...
+    match role_id {
+        consts::user_role::ROLE_ID_ORGANIZATION_ADMIN
+        | consts::user_role::ROLE_ID_INTERNAL_ADMIN
+        | consts::user_role::ROLE_ID_INTERNAL_VIEW_ONLY_USER => {
+            Err(UserErrors::InvalidDeleteOperation.into())
+                .attach_printable("Deletion not allowed for users with specific role id")
+        }
+        _ => Ok(()),
+    }
+}
```
Comment (racnan) [inline crates/router/src/utils/user.rs:126]: make this property of predefined permission.

**Review by racnan — CHANGES_REQUESTED** [review]

**Review by ivor-juspay — DISMISSED** [review]

**Review by ThisIsMani — APPROVED** [review]
LGTM

**Review by ivor-juspay — APPROVED** [review]

**Pull request merged**